game-core/src/main/java/games/strategy/engine/framework/startup/ui/pbem/ForumPosterEditor.java

```diff
@@ -0,0 +1,201 @@
+
+    row++;
+    add(
+        topicIdLabel,
+        new GridBagConstraints(
+            0,
+            row,
+            1,
+            1,
+            0,
+            0,
+            GridBagConstraints.WEST,
+            GridBagConstraints.NONE,
+            new Insets(0, 0, bottomSpace, labelSpace),
+            0,
+            0));
+
+    DocumentListenerBuilder.attachDocumentListener(
+        topicIdField, () -> viewModel.setTopicId(topicIdField.getText().trim()));
+    add(
+        topicIdField,
+        new GridBagConstraints(
+            1,
+            row,
+            1,
+            1,
+            1.0,
+            0,
+            GridBagConstraints.EAST,
+            GridBagConstraints.HORIZONTAL,
+            new Insets(0, 0, bottomSpace, 0),
+            0,
+            0));
+
+    viewPosts.addActionListener(e -> viewModel.viewForumButtonClicked());
+    add(
+        viewPosts,
+        new GridBagConstraints(
+            2,
+            row,
+            1,
+            1,
+            0,
+            0,
+            GridBagConstraints.EAST,
+            GridBagConstraints.NONE,
+            new Insets(0, 2, bottomSpace, 0),
+            0,
+            0));
+    row++;
+    add(
+        new JLabel(""),
+        new GridBagConstraints(
+            0,
+            row,
+            1,
+            1,
+            0,
+            0,
+            GridBagConstraints.WEST,
+            GridBagConstraints.NONE,
+            new Insets(0, 0, bottomSpace, labelSpace),
+            0,
+            0));
+    row++;
+
+    attachSaveGameToSummary.addChangeListener(
+        e -> viewModel.setAttachSaveGameToSummary(attachSaveGameToSummary.isSelected()));
+    add(
+        attachSaveGameToSummary,
+        new GridBagConstraints(
+            0,
+            row,
+            2,
+            1,
+            0,
+            0,
+            GridBagConstraints.WEST,
+            GridBagConstraints.NONE,
+            new Insets(0, 0, 0, 0),
+            0,
+            0));
+
+    testForum.addActionListener(e -> viewModel.testPostButtonClicked());
+    add(
+        testForum,
+        new GridBagConstraints(
+            2,
+            row,
+            1,
+            1,
+            0,
+            0,
+            GridBagConstraints.EAST,
+            GridBagConstraints.NONE,
+            new Insets(0, 0, 0, 0),
+            0,
+            0));
+    row++;
+    alsoPostAfterCombatMove.addActionListener(
+        e -> viewModel.setAlsoPostAfterCombatMove(alsoPostAfterCombatMove.isSelected()));
+    add(
+        alsoPostAfterCombatMove,
+        new GridBagConstraints(
+            0,
+            row,
+            2,
+            1,
+            0,
+            0,
+            GridBagConstraints.WEST,
+            GridBagConstraints.NONE,
+            new Insets(0, 0, 0, 0),
+            0,
+            0));
+  }
+
+  @Override
+  public void viewModelChanged(final ForumPosterEditorViewModel forumPosterEditorViewModel) {
```

Comment:
Notice that all other methods in this view class are removed, we just have a `viewModelChanged` method now.